File: .skills/openclaw-skills/skills/vnagin/zoom-manager-clawd/scripts/delete_meeting.js
#!/usr/bin/env node
const fs = require('fs').promises;
const path = require('path');
const fetch = require('node-fetch');

// Load config
const configPath = path.join(__dirname, '..', 'config.json');
let config;

async function loadConfig() {
    try {
        const data = await fs.readFile(configPath, 'utf8');
        config = JSON.parse(data);
    } catch (err) {
        console.error('Error loading config:', err.message);
        process.exit(1);
    }
}

async function getAccessToken() {
    const url = 'https://api.zoom.us/oauth/token';
    const params = new URLSearchParams({
        grant_type: 'account_credentials',
        account_id: config.account_id
    });

    try {
        const response = await fetch(url, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/x-www-form-urlencoded'
            },
            body: params.toString(),
            auth: `${config.client_id}:${config.client_secret}`
        });
        const data = await response.json();
        if (!response.ok) {
            throw new Error(`Failed to get token: ${data.reason || response.status}`);
        }
        return data.access_token;
    } catch (err) {
        console.error('Error getting access token:', err.message);
        process.exit(1);
    }
}

async function deleteMeeting(meetingId) {
    const token = await getAccessToken();
    const url = `https://api.zoom.us/v2/meetings/${meetingId}`;

    try {
        const response = await fetch(url, {
            method: 'DELETE',
            headers: {
                'Authorization': `Bearer ${token}`
            }
        });

        if (response.ok) {
            console.log(`Meeting ${meetingId} deleted successfully.`);
        } else {
            const error = await response.json();
            console.error(`Failed to delete meeting ${meetingId}: ${response.status} - ${error.message || 'Unknown error'}`);
        }
    } catch (err) {
        console.error('Error deleting meeting:', err.message);
    }
}

// Main execution
async function main() {
    await loadConfig();
    const meetingId = process.argv[2];
    if (!meetingId) {
        console.error('Usage: node delete_meeting.js <meeting_id>');
        process.exit(1);
    }
    await deleteMeeting(meetingId);
}

main().catch(console.error);

File: .skills/openclaw-skills/skills/vnagin/zoom-manager-clawd/scripts/delete_meeting.py
#!/usr/bin/env python3
import requests
import json
import os
import sys

# Загрузка конфигурации
try:
    with open('config.json', 'r') as f:
        config = json.load(f)
except FileNotFoundError:
    print("Error: config.json not found. Please create it with your Zoom API credentials.")
    sys.exit(1)

ACCOUNT_ID = config['account_id']
CLIENT_ID = config['client_id']
CLIENT_SECRET = config['client_secret']
USER_ID = config['user_id']

def get_access_token():
    url = "https://api.zoom.us/oauth/token"
    headers = {
        "Content-Type": "application/x-www-form-urlencoded"
    }
    data = {
        "grant_type": "account_credentials",
        "account_id": ACCOUNT_ID
    }
    response = requests.post(url, headers=headers, data=data, auth=(CLIENT_ID, CLIENT_SECRET))
    if response.status_code == 200:
        return response.json().get('access_token')
    else:
        print(f"Failed to get access token: {response.status_code} - {response.text}")
        return None

def delete_meeting(meeting_id):
    access_token = get_access_token()
    if not access_token:
        return False

    url = f"https://api.zoom.us/v2/meetings/{meeting_id}"
    headers = {
        "Authorization": f"Bearer {access_token}",
        "Content-Type": "application/json"
    }

    response = requests.delete(url, headers=headers)
    if response.status_code == 204:
        print(f"Meeting {meeting_id} deleted successfully.")
        return True
    else:
        print(f"Failed to delete meeting {meeting_id}: {response.status_code} - {response.text}")
        return False

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python delete_meeting.py <meeting_id>")
        sys.exit(1)

    meeting_id = sys.argv[1]
    delete_meeting(meeting_id)